File: cpp/leetcode/060.cpp
class Solution {
public:
    string getPermutation(int n, int k) {
        string res = "";
        string num = "123456789";
        vector<int> f(n, 1);
        for(int i = 1; i < n; i++) f[i] = f[i-1] * i;
        k--;
        for(int i = n - 1; i >= 0; i--){
            int j = k / f[i];
            k %= f[i];
            res += num.substr(j, 1);
            num.erase(j, 1);
        }
        return res;
    }
};

File: python/leetcode/060.py
class Solution(object):
    def getPermutation(self, n, k):
        factor, res = [1] * n, ""
        for i in range(1, n): factor[i] = factor[i-1] * i
        num = ['1', '2', '3', '4', '5', '6', '7', '8', '9']
        k -= 1
        for i in xrange(n, 0, -1):
            digit = num[k/factor[i-1]]
            res += str(digit)
            num.remove(digit)
            k %= factor[i-1]
        return res